assets/src/components/URLInput/URLInput.js

```diff
@@ -0,0 +1,50 @@
+import {Component} from '@wordpress/element';
+import {TextControl} from '@wordpress/components';
+
+export class URLInput extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+
+    const {__} = wp.i18n;
+    const { label, placeholder, value, onChange, disabled, help } = this.props
+
+    let errorMessage = "";
+
+
+    const isCustomLinkValid = () => {
```

Comment:
Probably best to use "URL" here for the naming as well.

From the naming you would not expect this function to have side effects and just return whether the url is valid. However it also changes `errorMessage`, which makes the logic a bit harder to follow.

Maybe it makes sense to have a `UrlValidationMessage` component instead? It would be a very simple component that just has the validation logic and only renders the `<span>` if it gets an invalid url. Then this `URLInput` component would also become simpler.